app/controllers/api/api_controller.rb

```diff
@@ -0,0 +1,23 @@
+module Api
+  class ApiController < ActionController::API
+    # layout false
+    def create
+      @user = User.find(params[:user_id])
+      @post = @user.posts.find(params[:post_id])
+      @comment = @post.comments.build(comment_params)
+      @comment.user = @user
+
+      if @comment.save
+        render json: @comment, status: :created
+      else
+        render json: { error: 'Error creating comment.' }, status: :unprocessable_entity
+      end
+    end
+
+    private
+
+    def comment_params
+      params.require(:comment).permit(:text)
+    end
```

Comment:
- [ ] why creating comments here in Api_controller, comments is already creating in comments controller, so kindly remove this